### PR TITLE
A few cleanup commits

### PR DESCRIPTION
**Pull request by brson**
## What have you changed? (mandatory)

- Remove the unused `Storage` from `tikv` root.
- Remove the renaming of `grpcio` to `grpc`. This looks like the remnants of an old refactoring.
- Turn on `clippy::nonminimal_bool`. This was disabled due to an old clippy bug that is fixed.

## What are the type of the changes? (mandatory)

- Misc (other changes)

## How has this PR been tested? (mandatory)

cargo check

## Does this PR affect documentation (docs) update? (mandatory)

no

## Does this PR affect tidb-ansible update? (mandatory)

no

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -122,6 +121,10 @@ kvproto = { git = "https://github.com/pingcap/kvproto.git" }
 log_wrappers = { path = "components/log_wrappers" }
 engine = { path = "components/engine" }
 
+[dependencies.grpcio]
```
Comment (siddontang) [inline Cargo.toml]: why not changing above uuid at the same time?

**Comment by siddontang** [comment]
Hi @brson 

Please fix the conflicts.

File: Cargo.toml (hunk 1 above)
Comment (brson) [reply Cargo.toml]: Can you rephrase the question? I don't understand.

Comment (brson) [reply Cargo.toml]: Oh, do you mean to write the uuid dependency the way the grpcio dependency is written?

I can do that, yes.

I shouldn't have reformatted grpcio though. It was an accident.

**Comment by brson** [comment]
Merges are fixed.

Instead of reformatting the uuid declaration in the manifest, I reverted the unnecessary change to the grpcio declaration.

**Comment by siddontang** [comment]
please fix the conflict @brson

**Comment by brson** [comment]
Re-merged.
